### ISSUE TITLE
Typings file not detected

### ISSUE THREAD

**Issue by jsundquist** [zurb/foundation-sites#11270]
## What should happen?
The typings file should be detected.

## What happens instead?
Receiving errors that the typing file cannot be found and should declare one

```
ERROR in src/client/foundation.ts(23,21):
TS7016: Could not find a declaration file for module 'foundation-sites/js/foundation.orbit'. 'node_modules/foundation-sites/js/foundation.orbit.js' implicitly has an 'any' type.
  Try `npm install @types/foundation-sites` if it exists or add a new declaration (.d.ts) file containing `declare module 'foundation-sites';
```

## Possible Solution

Unknown. Typings file does exist but not being picked up for some reason.

## Test Case and/or Steps to Reproduce

Test Case: Unable to provide a valid test case using code pen.

How to reproduce:
1. Create a file called `foundation.ts`
2. add the line `import {Foundation} from 'foundation-sites/js/foundation.core';`
3. Note the ide identifies that the typings definitions file is not found.

## Context

Typescript is throwing errors though it does not officially cause a build to break. Ideally no errors should be thrown.

## Your Environment

- Foundation version(s) used: 6.4.4-RC1
- Browser(s) name and version(s): N/a
- Operating System and version (desktop or mobile): Desktop MacOs 10.13.4
- Node: 8.11.1
- npm: 5.6.0
- yarn: 1.6.0
- typescript: 2.8.3

## Checklist (all required):

- [X] I have read and follow the CONTRIBUTING.md document.
- [X] There are no other issues similar to this one.
- [X] The issue title is descriptive.
- [X] The template is correctly filled.

**Comment by jsundquist** [comment]
One thing I just noticed that if I import `import Foundation from 'foundation-sites` and manual modify the dist/js/foundation.d.ts file so it has 

```
declare module 'foundation-sites' {
  export Foundation
}
```

instead of the original 
```
declare module Foundation {
  export = Foundation;
}
```

the above line does work. however if you try to import specific packages vs the world,  there are still type definition errors.

**Comment by DanielRuf** [comment]
Do you know what we would have to change to get it properly working?

cc @samvloeberghs

**Comment by samvloeberghs** [comment]
hi @DanielRuf I havent been involved in the typings for a while.
Normally these days typings should be available in the module itself:

https://github.com/zurb/foundation-sites/blob/develop/dist/js/foundation.d.ts

**Comment by DanielRuf** [comment]
@kball @DaSchTour @Anidetrix do we need some specific field in package.json or some special mappings?

**Comment by DaSchTour** [comment]
Well. Typings are linked from package.json from the typings property. And from what I can see everything looks good. But I've to admit I haven't used foundation JS for ages. So I wonder a bit what has changed since introduction of typings here and what has changed with 6.4.4

**Comment by DanielRuf** [comment]
This should be resolved with the latest releases.
Can anybode test this?

**Comment by hvolmer** [comment]
Hello. I am trying to import Foundation into an Angular 9 app today, and getting the error that ".../foundation.d.ts is not a module.  If I modify the d.ts to

declare module 'foundation-sites' {
  export = Foundation
}

then I can get things working.

Foundation-sites v6.6.3
Node 12.17.0
Typescript 3.8.3

**Comment by DanielRuf** [comment]
> Hello. I am trying to import Foundation into an Angular 9 app today, and getting the error that ".../foundation.d.ts is not a module. If I modify the d.ts to
> 
> declare module 'foundation-sites' {
> export = Foundation
> }

This is probably because we ship CJS and ESM bundles and your setup probably uses the ESM version. In this case we might need both declarations. Can you check if this also works? One for the module and a normal / the original declaration?

**Comment by hvolmer** [comment]
> > Hello. I am trying to import Foundation into an Angular 9 app today, and getting the error that ".../foundation.d.ts is not a module. If I modify the d.ts to
> > declare module 'foundation-sites' {
> > export = Foundation
> > }
> 
> This is probably because we ship CJS and ESM bundles and your setup probably uses the ESM version. In this case we might need both declarations. Can you check if this also works? One for the module and a normal / the original declaration?

This appears to work, if this is what you are suggesting:

`declare module "foundation-sites" {
  export = Foundation;
}
declare module "Foundation" {
  export = Foundation;
}`

and then I can:

`import { Reveal } from 'foundation-sites';`

**Comment by DanielRuf** [comment]
Great, can you provide a PR with this change? This would be very helpful and great.

**Comment by hvolmer** [comment]
I assume that what I need to actually edit for a PR is 

foundation-sites/js/typescript/foundation.d.ts

And something/someone else will take care of the build?  Sorry, I'm haven't contributed to public repos much or at all. Thanks!

**Comment by DanielRuf** [comment]
> I assume that what I need to actually edit for a PR is
> 
> foundation-sites/js/typescript/foundation.d.ts

Correct. Create a separate branch, make the changes, push them to your branch on the forked repo and then open a new PR.

**Comment by paxperscientiam** [comment]
Did this PR ever happen?
